reports/glides_terminal_missed_extraneous_predictions.livemd

```diff
@@ -421,39 +467,62 @@ Methodology:
 * From VehiclePositions, get all timestamps (truncated to minute) at which a vehicle actually departed a stop.[^1]
 * From TripUpdates, get all timestamps (truncated to minute) at which a vehicle was predicted to depart a stop.[^2]
 * If a vehicle actually departed stop S at the same minute that a vehicle was predicted to depart stop S, then that prediction is considered accurate.
+* Also compute the above, but with predictions and departures compared by trip ID instead of timestamp.
 
 ---
 
 [^1]: There is no "departing stop" vehicle status, so we look for events where the vehicle is "IN_TRANSIT_TO" or "INCOMING_AT" the stop _after_ the target stop.
 [^2]: This is the set of _all_ times at which a vehicle was predicted to depart a stop. If at any moment, even just for a minute, a vehicle was predicted to depart stop S at time T, then that `{time, stop}` pair is added to the set.
 
 ```elixir
-table =
+times_table =
   0..23
   # Service day starts at 4am, so let's start the table at that hour.
   |> Enum.map(&rem(&1 + 4, 24))
   |> Enum.map(fn hour ->
-    predicted_departure_time_count =
-      predicted_departure_times_by_hour
+    predicted_time_count =
+      predicted_times_by_hour
       |> Map.get(hour, MapSet.new())
       |> MapSet.size()
 
-    actual_departure_time_count =
-      actual_departure_times_by_hour
+    actual_time_count =
+      actual_times_by_hour
       |> Map.get(hour, MapSet.new())
       |> MapSet.size()
 
-    percentage = Map.fetch!(predicted_percentages_by_hour, hour)
+    time_percentage = Map.fetch!(predicted_time_percentages_by_hour, hour)
 
     [
       {"hour", "#{GlidesReport.Util.zero_pad(hour)}:00"},
-      {"# of predicted departure times", predicted_departure_time_count},
-      {"# of actual departure times", actual_departure_time_count},
-      {"% of actual departure times that were also predicted", percentage}
+      {"# of predicted departure times", predicted_time_count},
+      {"# of actual departure times", actual_time_count},
+      {"% of actual departure times that were also predicted", time_percentage}
     ]
   end)
 
-table_name = "Predicted vs actual departures"
+trips_table =
+  0..23
+  |> Enum.map(&rem(&1 + 4, 24))
```

Comment:
Why `+ 4`?

Oop, I see this above:
`# Service day starts at 4am, so let's start the table at that hour.`

Feels nitpicky but can we move that here as well, in case this block of code is ever moved? (Or add a named function but that seems unnecessary)

Reply:
Sure!